hooks/read-package-hook/src/createPeerDependencyPatcher.ts

```diff
@@ -36,11 +36,10 @@ export function createPeerDependencyPatcher (
         pkg.peerDependencies![peerName] = '*'
         continue
       }
-      if (
-        !peerDependencyRules.allowedVersions?.[peerName] ||
```

Comment:
The original test cases for this feature had `allowedVersions` defined with both a `child` selector and `parent>child` selector. That test case didn't appropriately test this continue branch logic.